### PR TITLE
brewtarget: init at 2.3.1

### PR DESCRIPTION
**Pull request by mmahut**
###### Motivation for this change

init at 2.3.1

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by worldofpeace — CHANGES_REQUESTED** [review]
You need to package this as described in the `qt` section of the nixpkgs manual
* https://hydra.nixos.org/build/96645877/download/1/nixpkgs/manual.html#sec-language-qt

Note that's the unstable build because it's changed on master.

**Comment by mmahut** [comment]
@worldofpeace thank you for your suggestion! Rebased.

File: pkgs/applications/misc/brewtarget/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ lib,
+mkDerivation,
```
Comment (worldofpeace) [inline pkgs/applications/misc/brewtarget/default.nix]: I always find the way the qt manual wants to suggest this style not great.

```
{ lib
, mkDerivation
, ...
```

is what's already suggested in the "style guide".

File: pkgs/applications/misc/brewtarget/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "Brewtarget";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "14xmm6f8xmvypagx4qdw8q9llzmyi9zzfhnzh4kbbflhjbcr7isz";
+  };
+
+  nativeBuildInputs = [ cmake wrapQtAppsHook ];
```
Comment (worldofpeace) [inline pkgs/applications/misc/brewtarget/default.nix]:
```suggestion
  nativeBuildInputs = [ cmake ];
```

File: pkgs/applications/misc/brewtarget/default.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+wrapQtAppsHook,
+qtbase,
+qttools,
+qtmultimedia,
+qtwebkit,
+qtsvg
+}:
+
+let
+  pname = "brewtarget";
```
Comment (worldofpeace) [inline pkgs/applications/misc/brewtarget/default.nix]: does it not accept `pname` and `version`?

maybe you're missing the `rec` in `mkDerivation rec { ...` so we can self reference.

**Comment by mmahut** [comment]
@worldofpeace rebased.

**Review by worldofpeace — APPROVED** [review]
LGTM, builds and executes. Traversed the output directory structure.

**Pull request merged**